modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -25,18 +25,42 @@
 package com.sun.glass.utils;
 
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.IOException;
 import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.security.AccessController;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.security.PrivilegedAction;
+import java.util.Arrays;
 import java.util.HashSet;
+import java.util.List;
 
 public class NativeLibLoader {
 
     private static final HashSet<String> loaded = new HashSet<String>();
+    private static final String tmpdir = System.getProperty("java.io.tmpdir");
```

Comment:
`tmpdir` is unused and can be removed

Reply:
No, actually we need to fall back on "java.io.tmpdir" when "user.home" is not writable, for whatever reason, it happens. BTW, it would also be nice to let users set the cache directory.